src/lib/snack-bar/snack-bar-container.ts

```diff
@@ -0,0 +1,46 @@
+import {
+  Component,
+  ComponentRef,
+  ViewChild
+} from '@angular/core';
+import {
+  BasePortalHost,
+  ComponentPortal,
+  TemplatePortal,
+  PortalHostDirective} from '@angular2-material/core';
+import {MdSnackBarConfig} from './snack-bar-config';
+import {MdSnackBarContentAlreadyAttached} from './snack-bar-errors';
+
+
+/**
+ * Internal component that wraps user-provided snack bar content.
+ */
+@Component({
+  moduleId: module.id,
+  selector: 'snack-bar-content',
+  templateUrl: 'snack-bar-container.html',
+  styleUrls: ['snack-bar-container.css'],
+  host: {
+    '[attr.role]': 'snackBarConfig?.role'
+  }
+})
+export class MdSnackBarContainer extends BasePortalHost {
+  /** The portal host inside of this container into which the snack bar content will be loaded. */
+  @ViewChild(PortalHostDirective) private _portalHost: PortalHostDirective;
```

Comment:
A property with `@ViewChild` can't be private because Angular has to access it. (it will be an error in AoT). You can leave the underscore, though, to imply that it is "internal".